src/imgtools/logging/__init__.py

```
@@ -0,0 +1,223 @@
+
+		processors = [
+			*pre_chain,
+			structlog.processors.StackInfoRenderer(),
+		]
+
+		if self.json_logging:
+			log_dir = self.base_dir / '.imgtools' / 'logs'
+			try:
+				log_dir.mkdir(parents=True, exist_ok=True)
+			except PermissionError as e:
+				raise RuntimeError(f"Cannot create log directory at {log_dir}: {e}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)